buildSrc/src/main/groovy/org/elasticsearch/hadoop/gradle/BuildPlugin.groovy

```diff
@@ -590,72 +582,52 @@ class BuildPlugin implements Plugin<Project>  {
      * @param project to be configured
      */
     private static void configureIntegrationTestTask(Project project) {
-        Jar hadoopTestingJar = project.rootProject.tasks.findByName('hadoopTestingJar') as Jar
-        if (hadoopTestingJar == null) {
-            // jar used for testing Hadoop remotely (es-hadoop + tests)
-            hadoopTestingJar = project.rootProject.tasks.create('hadoopTestingJar', Jar)
-            hadoopTestingJar.dependsOn(project.rootProject.tasks.getByName('jar'))
-            hadoopTestingJar.classifier = 'testing'
-            project.logger.info("Created Remote Testing Jar")
-        }
-
-        // Add this project's classes to the testing uber-jar
-        hadoopTestingJar.from(project.sourceSets.test.output)
-        hadoopTestingJar.from(project.sourceSets.main.output)
-        hadoopTestingJar.from(project.sourceSets.itest.output)
-
-        Test integrationTest = project.tasks.create('integrationTest', RestTestRunnerTask.class)
-        integrationTest.dependsOn(hadoopTestingJar)
-
-        integrationTest.testClassesDirs = project.sourceSets.itest.output.classesDirs
-        integrationTest.classpath = project.sourceSets.itest.runtimeClasspath
-        integrationTest.excludes = ["**/Abstract*.class"]
-
-        integrationTest.ignoreFailures = false
-
-        integrationTest.executable = "${project.ext.get('runtimeJavaHome')}/bin/java"
-        integrationTest.minHeapSize = "256m"
-        integrationTest.maxHeapSize = "2g"
-
-        integrationTest.testLogging {
-            displayGranularity 0
-            events "started", "failed" //, "standardOut", "standardError"
-            exceptionFormat "full"
-            showCauses true
-            showExceptions true
-            showStackTraces true
-            stackTraceFilters "groovy"
-            minGranularity 2
-            maxGranularity 2
-        }
+        if (project != project.rootProject) {
+            Jar itestJar = project.tasks.create('itestJar', Jar)
+            itestJar.dependsOn(project.tasks.getByName('jar'))
```

Comment:
Why do we need the regular JAR? We are just packaging the sourceset output, I don't actually need to bundle the normal JAR to create this one.

Reply:
Hadoop maintains a separate classpath that needs to be configured for each job in the driver program. We use the local job runners in the integration tests, but they still use their own classloaders when running the local job.

Historically, we configured the job jar (which contains the code for the integ test jobs, as well as the project code being tested) once on the test config. I'd like to move to using the original jar as-is without uber-jarring it with the itest code, but that will require updating every test case since it needs to be configured on every test job (can't be set on the configuration once and be done).